python/lsst/ctrl/execute/slurmPlugin.py

```diff
@@ -188,6 +187,180 @@ def createAllocationFile(self, input):
         os.chmod(outfile, 0o755)
         return outfile
 
+    def largeGlideinsFromJobPressure(self, generatedSlurmFile):
+        """Calculate the number of glideins needed from job pressure
+
+        Returns
+        -------
+        number : `str`
+            The number of glideins
+        """
+
+        import hashlib
+        import socket
+        import time
+
+        import htcondor
+        from lsst.ctrl.bps.htcondor import condor_q
+
+        verbose = self.isVerbose()
+
+        autoCPUs = self.getAutoCPUs()
+        memoryPerCore = self.getMemoryPerCore()
+        auser = self.getUserName()
+
+        try:
+            schedd_name = socket.getfqdn()
+            coll = htcondor.Collector()
+            schedd_ad = coll.locate(htcondor.DaemonTypes.Schedd)
+            scheddref = htcondor.Schedd(schedd_ad)
+
+            # Query for jobs that have (bps_run isnt Undefined)
+            #                          (bps_job_label isnt Undefined)
+            # and amongst those identify Large jobs
+            #                          thisCores > 16
+            #                          thisMemory > 16*4096
+            #
+            # projection contains the job classads to be returned.
+            # These include the cpu and memory profile of each job,
+            # in the form of RequestCpus and RequestMemory
+            projection = [
+                "bps_run",
+                "bps_job_label",
+                "JobStatus",
+                "Owner",
+                "RequestCpus",
+                "JobUniverse",
+                "RequestMemory",
+            ]
+            owner = f'(Owner=="{auser}") '
+            jstat = "&& (JobStatus==1) "
+            bps1 = "&& (bps_run isnt Undefined) "
+            bps2 = "&& (bps_job_label isnt Undefined) "
+            juniv = "&& (JobUniverse==5)"
+            # The constraint determines that the jobs to be returned belong to
+            # the current user (Owner) and are Idle vanilla universe jobs.
+            full_constraint = f"{owner}{jstat}{bps1}{bps2}{juniv}"
+            if verbose:
+                print("Find Large BPS Jobs:")
```

Comment:
True, it is not "Find Large BPS Jobs:" on the very next line, but it is, in the flow of things, that is what is occurring now in the code that follows.   I'll see if that can be part of the query, but it is an Or of sorts, large Memory or high number of cpus, so I'll see.

Reply:
ok I do see that my "test suite" of jobs had Large bps jobs, and small not bps jobs, and so the 
statement 
           if len(condorq_data) == 0:
                print("No Large BPS Jobs.")
would not be correct "yet", it would have to be lower after the filtering.

Reply:
I believe the query of the schedd (searching for Large jobs) for job classads is now efficient with RequestMemory, RequestCpus constraints included, so there is no filtering after the fact.